src/hotspot/share/runtime/os.hpp

```diff
@@ -223,6 +223,10 @@ class os: AllStatic {
   static bool   pd_unmap_memory(char *addr, size_t bytes);
   static void   pd_free_memory(char *addr, size_t bytes, size_t alignment_hint);
   static void   pd_realign_memory(char *addr, size_t bytes, size_t alignment_hint);
+  static void   pd_pretouch_memory(void *start, void *end, size_t page_size);
+
+  // Fallback to this if OS needs no specific treatment
+  static void   pretouch_memory_fallback(void *start, void *end, size_t page_size);
```

Comment:
Would `pretouch_memory_common` be a better name for this method?  The name should reflect why the method exists, and I think of this as common code to avoid having a copy of the implementation in each platform-specific class.  You don't "fall back" to this implementation: on most platforms, this is the only implementation.  If `class os` had virtual methods, this would be the base class implementation.  

(Not a big deal.)